src/relay/connection.ts

```diff
@@ -276,33 +281,48 @@ class RelayConnection extends EventEmitter implements MultiaddrConnection {
     while (true) {
       let promises: Promise<SinkType>[] = []
 
+      let resolvedPromiseName
```

Comment:
why don't use it for the conditions below? By doing this, we can get rid of a lot of state management complexity